client/x/evmengine/keeper/abci.go

```diff
@@ -280,7 +281,7 @@ func (k *Keeper) startBuild(ctx context.Context, feeRecipient common.Address, wi
 
 	attrs := &engine.PayloadAttributes{
 		Timestamp:             ts,
-		Random:                head.Hash(), // We use head block hash as randao.
+		Random:                crypto.RandomHash(),
```

Comment:
shall we guarantee the deterministic based on the proposer?

Reply:
What would be required for that? 

I think some kind of determinism makes randomness not random enough, if that makes sense

Reply:
This is how it's solved in Prism (ETH CL client)
https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/core/blocks/randao.go

Value is derivated from beacon state and validator's key.

https://eips.ethereum.org/EIPS/eip-4399
> Obviously, historical randomness provided by any decentralized oracle is 100% predictable.
> Make your applications rely on the future randomness with a reasonably high lookahead

Use of RANDAO is not very extended for applications, I think it's better to have this be a known issue that we document, rather than rushing a solution.